CONTRIBUTING.md

```diff
@@ -1,10 +1,40 @@
-You want to contribute? We really appreciate that.
-Please have a look at this list to facilitate your contribution:
+# Want to contribute some code? 
+
+## We’d really appreciate your support!
+
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Focus on and improve the strengths of Jamulus
+There are other solutions for real time online jamming available. But Jamulus stands alone with its client/server system, minimalistic GUI and simple functionality. 
```

Comment:
This is the "Why Jamulus rather than ..?" quiz :).

I've not heard much comparative discussion of Jamulus against other realtime jamming solutions in terms of the user experience, either - at least, not by dispassionate parties without specific allegiances...

This is written to the developer, though.  But even there, "simple" it isn't...  I think this was trying to say that the aim was "Run it, click connect, jam" - simple.  Just not saying "Run in.  Work out what Soundcard Device I'm using.  Work out why it doesn't like it.  Work out whether I need to worry about all those other settings.  Click connect.  Wait.  Why did only a handful of servers and not the one I was told to pick show up?"

;)

Reply:
It feels like a re-statement of the "KISS" principle in the sense that it emphasises minimalism and simplicity. So either we remove it as overlapping too much, or think of another principle that makes Jamulus special. But as you say, nothing immediately comes to mind.

Reply:
I'd suggest removing it for now. It will be hard enough to get contributors to read the whole document. It'll be simpler if it's shorter and avoids redundancy.

Reply:
OK. And if we think of any other principle to live by, we can always add it.